### PR TITLE
Support HOCON format

### PR DESCRIPTION
**Pull request by alinadz720**

### CODE REVIEW

**Review by vstyran — CHANGES_REQUESTED** [review]

File: src/main/java/org/mdcfg/source/HoconSource.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        return folder.listFiles((dir, name) -> name.toLowerCase().endsWith(".conf"));
+    }
+
+    private String numberRows(String sourceData) {
+        Matcher matcher = pattern.matcher(sourceData);
+        StringBuilder res = new StringBuilder();
+        int rowNumber = 0;
+
+        while(matcher.find()) {
+            matcher.appendReplacement(res, rowNumber + matcher.group());
```
Comment (vstyran) [inline src/main/java/org/mdcfg/source/HoconSource.java:58]: insert invisible ACII symbol between number and selector

File: src/main/java/org/mdcfg/source/HoconSource.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class HoconSource extends FileSource {
+
+    private static final TypeReference<Map<String, Object>> TYPE = new TypeReference<>() {};
+    private final Pattern pattern = Pattern.compile("[a-zA-Z]*@");
```
Comment (vstyran) [inline src/main/java/org/mdcfg/source/HoconSource.java:22]: format of selector could be     add-in*@[ls, panoramic-roof]: 55000

**Review by vstyran — CHANGES_REQUESTED** [review]

File: src/main/java/org/mdcfg/source/HoconSource.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            matcher.appendReplacement(res, "\"" + rowNumber + MARKER + matcher.group(1));
+            rowNumber++;
+        }
+        matcher.appendTail(res);
+        return res.toString();
+    }
+
+    private Map<String, Object> orderData(Map<String, Object> rawData) {
+        return rawData.entrySet().stream()
+                .sorted(Comparator.comparing(e -> Integer.parseInt(e.getKey().substring(0, e.getKey().indexOf(MARKER)))))
```
Comment (vstyran) [inline src/main/java/org/mdcfg/source/HoconSource.java:68]: may be split by marker then sort by splitt[0]  and then map[1] is better by performance

File: src/main/java/org/mdcfg/source/HoconSource.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                .collect(Collectors.toMap(e -> e.getKey().substring(e.getKey().indexOf(MARKER) + 1),
```
Comment (vstyran) [inline src/main/java/org/mdcfg/source/HoconSource.java:69]: and test with subkeys, probably will crash

**Review by aolyvko — APPROVED** [review]

**Pull request merged**